unison-core/src/Unison/Names.hs

```diff
@@ -381,10 +381,13 @@ difference a b = Names (R.difference (terms a) (terms b))
                        (R.difference (types a) (types b))
 
 contains :: Names -> Reference -> Bool
-contains names r =
-  -- this check makes `contains` O(n) instead of O(log n)
-  (Set.member r . Set.map Referent.toReference . R.ran) (terms names)
-  || R.memberRan r (types names)
+contains names =
+  \r -> Set.member r termsReferences || R.memberRan r (types names)
```

Comment:
Please comment why this lambda is intentional or it's likely someone will come along, see a "redundant" lambda and undo your performance optimization here, I don't generally think of this sort of optimization unless specifically prompted to and I suspect others don't either.